src/pages/auth/Register.tsx

```diff
@@ -21,89 +72,157 @@ const Register = () => {
   const submitHandler = async (event: any) => {
     event.preventDefault();
 
-    if (password !== password2) {
-      setError({
-        error: true,
-        message: "Both passwords should match",
-      });
-      return;
-    } else {
-      setError({
-        error: false,
-        message: "",
-      });
-    }
-
-    const user = {
-      firstName,
-      lastName,
-      email,
-      password,
-    };
+    if (
+      firstName.length > 0 &&
+      lastName.length > 0 &&
+      email.length > 0 &&
+      password.length > 0 &&
+      password2.length > 0
+    ) {
+      if (password !== password2) {
+        setError({
+          error: true,
+          message: "Both passwords should match",
+        });
+        return;
+      } else {
+        setError({
+          error: false,
+          message: "",
+        });
+      }
 
-    const req = await fetch("http://localhost:5001/api/auth/register", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(user),
-    });
+      const user = {
+        firstName,
+        lastName,
+        email,
+        password,
+      };
 
-    const res = await req.json();
-    console.log(res);
-    if (res.success) {
-      setError({
-        error: false,
-        message: res.message,
+      const req = await fetch("http://localhost:5001/api/auth/register", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(user),
       });
 
-      window.location.href = "/landing/signin";
+      const res = await req.json();
+      if (res.success) {
+        setError({
+          error: false,
+          message: res.message,
+        });
+
+        window.location.href = "/beneficiaries";
```

Comment:
Rather than redirecting using the window object (which we want to avoid in React, as it's a single-page app), use the react-router `useNavigate` hook.